lib/chef/knife/core/generic_presenter.rb

```diff
@@ -182,7 +182,7 @@ def extract_nested_value(data, nested_value_spec)
             #   `keys` - want the key named `keys`, not a list of
             #   available keys.
             elsif data.respond_to?(:[])
-              data = data[attr]
+              data = data[attr] ? data[attr] : data.send(attr.to_sym)
```

Comment:
Looks like this is basically duplicating the next `elsif` clause ( `elsif data.respond_to?(attr.to_sym)` ). Would it be better to change the `elsif` clause on line 184 to `data.respond_to?(:[]) && data.has_key?(attr)` ?